scripts/validate_docstrings.py

```diff
@@ -436,16 +436,20 @@ def signature_parameters(self):
                 # accessor classes have a signature but don't want to show this
                 return tuple()
         try:
-            sig = inspect.getfullargspec(self.obj)
+            sig = inspect.signature(self.obj)
         except (TypeError, ValueError):
             # Some objects, mainly in C extensions do not support introspection
             # of the signature
             return tuple()
-        params = sig.args
-        if sig.varargs:
-            params.append("*" + sig.varargs)
-        if sig.varkw:
-            params.append("**" + sig.varkw)
+        params = list()
+        for parameter, data in sig.parameters.items():
+            if data.kind == inspect.Parameter.VAR_POSITIONAL:
+                params.append("*" + parameter)
+            elif data.kind == inspect.Parameter.VAR_KEYWORD:
+                params.append("**" + parameter)
+            else:
+                params.append(parameter)
```

Comment:
I think this would be more readable if we implement a function like:
```python
def with_stars(name, param_info):
    if param_info.kind == inspect.Parameter.VAR_POSITIONAL:
        return '*{}'.format(name)
    ...
    return name

params = [with_stars(name, param_info) for name, param_info in sign.parameters.items())]
```